app/src/main/java/com/github/wnder/TakePictureActivity.java

```diff
@@ -0,0 +1,131 @@
+import android.provider.MediaStore;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class TakePictureActivity extends AppCompatActivity {
+    private Button takePictureButton;
+    private Button pictureConfirmButton;
+    private ImageView picture;
+    private final int TAKE_PHOTO = 0;
+    private String currentPhotoPath;
+    private String imageFileName;
+    private Bitmap currentBitmap;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_take_picture);
+        picture = findViewById(R.id.imageFromCamera);
+
+        takePictureButton = findViewById(R.id.takePictureButton);
+        takePictureButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                dispatchTakePictureIntent();
+            }
+        });
+
+        pictureConfirmButton = findViewById(R.id.pictureConfirmButton);
+        pictureConfirmButton.setVisibility(View.INVISIBLE);
+        pictureConfirmButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                storeBitmapInGallery();
+            }
+        });
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent cameraIntent){
+        super.onActivityResult(requestCode, resultCode, cameraIntent);
+        if(resultCode == RESULT_OK && requestCode == TAKE_PHOTO){
+            setPic();
+            pictureConfirmButton.setVisibility(View.VISIBLE);
+        }
+    }
+
+    private void storeBitmapInGallery(){
+        MediaStore.Images.Media.insertImage(getContentResolver(), currentBitmap, imageFileName , "");
+    }
+
+    private File createImageFile() throws IOException {
+        // Create an image file name
+        String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
+        imageFileName = "JPEG_" + timeStamp + "_";
+        File storageDir = getExternalFilesDir(Environment.DIRECTORY_PICTURES);
+        File image = File.createTempFile(
+                imageFileName,  /* prefix */
+                ".jpg",         /* suffix */
+                storageDir      /* directory */
+        );
+
+        // Save a file: path for use with ACTION_VIEW intents
+        currentPhotoPath = image.getAbsolutePath();
+        return image;
+    }
+
+    private void dispatchTakePictureIntent() {
+        Intent takePictureIntent = new Intent(android.provider.MediaStore.ACTION_IMAGE_CAPTURE);
+        // Create the File where the photo should go
+        File photoFile = null;
+        try {
+            photoFile = createImageFile();
+        } catch (IOException ex) {
+            // Error occurred while creating the File
```

Comment:
Shouldn't we at least print that something went wrong if we catch an error? I'm not sure of what is the proper way of doing "try catch" for exception on android though